src/app/api/imageInput/route.ts

```diff
@@ -0,0 +1,158 @@
+import { ChatOpenAI } from "@langchain/openai";
+import { HumanMessage } from "@langchain/core/messages";
+import { env } from "@/app/env.mjs";
+import { NextResponse } from "next/server";
+import { saveDroppedImage } from "@/utils/apiHelper";
+import { saveToDB } from "@/utils/apiHelper";
+import { z } from "zod";
+import { Message } from "ai/react/dist";
+import { auth } from "@clerk/nextjs";
+
+const dropZoneInputSchema = z.object({
+  imageName: z.string(),
+  imageType: z.string(),
+  imageSize: z.number(),
+  value: z.string(),
+  userId: z.string(),
+  orgId: z.string(),
+  chatId: z.any(),
+  id: z.string(),
+  imageFile: z.any(),
+  messages: z.any(),
+});
+const chat = new ChatOpenAI({
+  modelName: "gpt-4-vision-preview",
+  maxTokens: 1024,
+  openAIApiKey: env.OPEN_AI_API_KEY,
+});
+
+export async function POST(request: Request) {
+  const formData = await request.formData();
+  const zodMessageString: any = formData.get("zodMessage");
+  const file: any = formData.get("file");
+  const zodMessageObject = JSON.parse(zodMessageString);
+
+  const zodMessage = dropZoneInputSchema.parse({
+    imageName: zodMessageObject.data.imageName,
+    imageType: zodMessageObject.data.imageType,
+    imageSize: zodMessageObject.data.imageSize,
+    value: zodMessageObject.data.value,
+    userId: zodMessageObject.data.userId,
+    orgId: zodMessageObject.data.orgId,
+    chatId: zodMessageObject.data.chatId,
+    id: zodMessageObject.data.id,
+    imageFile: file,
+    messages: zodMessageObject.data.message as Message[],
+  });
+
+  // console.log("zodMessage:", zodMessage);
+
+  const {
+    imageName,
+    imageType,
+    imageSize,
+    value,
+    userId,
+    orgId,
+    chatId,
+    id,
+    messages,
+    imageFile,
+  } = zodMessage;
+  const { orgSlug } = await auth();
+
+  const _message = messages as unknown as Message[];
+  // console.log("_message", _message);
+  const url = request.url;
+  const urlArray = url.split("/");
+
+  if (!formData || !formData.has("file")) {
+    return NextResponse.json(
+      { result: "no data received", success: false },
+      { status: 400 },
+    );
+  }
+  const parts = imageFile.name.split(".");
+  const extension = parts[parts.length - 1];
+  // console.log("messages",messages)
+  const formattedMessages = messages.map((msg: any) => {
+    return {
+      role: msg.role,
+      content: msg.content,
+    };
+  });
+  // console.log("formatemsgs", formattedMessages);
+  const combinedContent = formattedMessages
+    .map((msg: any) => msg.content)
+    .join("\n");
+  const combinedMessage = {
+    role: "user",
+    content: combinedContent,
+  };
```

Comment:
we decided not to use a single message instead you have to feed all the message to model to get proper context.

Reply:
older one